### PR TITLE
[irods/irods#7266] Add Debian 12 support

### PR DESCRIPTION
**Pull request by alanking**
In service of irods/irods#7266

Running tests soon...

### CODE REVIEW

File: projects/debian-12/release.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/*
+
+RUN python3 -m pip install 'unittest-xml-reporting<3.1.0'
+
+RUN mkdir -p /irods_testing_environment_mount_dir && chmod 777 /irods_testing_environment_mount_dir
+
+ENTRYPOINT ["bash", "-c", "until false; do sleep 2147483647d; done"]
+
+ARG irods_package_version=4.3.0-1~bullseye
```
Comment (korydraughn) [inline projects/debian-12/release.Dockerfile:41]: `bullseye` -> `bookworm`?

Comment (alanking) [reply projects/debian-12/release.Dockerfile:41]: Indeed

**Comment by alanking** [comment]
Unit tests passed. Running core tests

**Comment by alanking** [comment]
Aaand core tests passed. Except one that I just wrote and isn't finished that I forgot about. But it cleaned up and everything else passed.

**Comment by trel** [comment]
Amazing.   Ship it.

**Pull request merged**